applet/src/main/java/jpake/jpakePassivePayload.java

```diff
@@ -0,0 +1,213 @@
+        short encAlen = input[currStart];
+        currStart += 1;
+        byte[] encA = new byte[encAlen];
+        Util.arrayCopyNonAtomic(input, currStart, encA, (short)0, encAlen);
+        ECPoint A = jpakeActor.curve.decodePoint(encA);
+        currStart += encAlen;
+        short pubA1len = input[currStart];
+        currStart += 1;
+        byte[] encPublicA1 = new byte[pubA1len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicA1, (short)0, pubA1len);
+        ECPoint publicA1 = jpakeActor.curve.decodePoint(encPublicA1);
+        currStart += pubA1len;
+        short pubV1len = input[currStart];
+        currStart += 1;
+        byte[] encPublicV1 = new byte[pubV1len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicV1, (short)0, pubV1len);
+        ECPoint publicV1 = jpakeActor.curve.decodePoint(encPublicV1);
+        currStart += pubV1len;
+        short result1len = input[currStart];
+        currStart += 1;
+        byte[] encResult1 = new byte[result1len];
+        Util.arrayCopyNonAtomic(input, currStart, encResult1, (short)0, result1len);
+        BigInteger result1 =  new BigInteger(encResult1);
+        currStart += result1len;
+        short pubA2len = input[currStart];
+        currStart += 1;
+        byte[] encPublicA2 = new byte[pubA2len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicA2, (short)0, pubA2len);
+        ECPoint publicA2 = jpakeActor.curve.decodePoint(encPublicA2);
+        currStart += pubA2len;
+        short pubV2len = input[currStart];
+        currStart += 1;
+        byte[] encPublicV2 = new byte[pubV2len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicV2, (short)0, pubV2len);
+        ECPoint publicV2 = jpakeActor.curve.decodePoint(encPublicV2);
+        currStart += pubV2len;
+        short result2len = input[currStart];
+        currStart += 1;
+        byte[] encResult2 = new byte[result2len];
+        Util.arrayCopyNonAtomic(input, currStart, encResult2, (short)0, result2len);
+        BigInteger result2 =  new BigInteger(encResult2);
+        currStart += result2len;
+        short pubA3len = input[currStart];
+        currStart += 1;
+        byte[] encPublicA3 = new byte[pubA3len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicA3, (short)0, pubA3len);
+        ECPoint publicA3 = jpakeActor.curve.decodePoint(encPublicA3);
+        currStart += pubA3len;
+        short pubV3len = input[currStart];
+        currStart += 1;
+        byte[] encPublicV3 = new byte[pubV3len];
+        Util.arrayCopyNonAtomic(input, currStart, encPublicV3, (short)0, pubV3len);
+        ECPoint publicV3 = jpakeActor.curve.decodePoint(encPublicV3);
+        currStart += pubV3len;
+        short result3len = input[currStart];
+        currStart += 1;
+        byte[] encResult3 = new byte[result3len];
+        Util.arrayCopyNonAtomic(input, currStart, encResult3, (short)0, result3len);
+        BigInteger result3 =  new BigInteger(encResult3);
+
```

Comment:
Check if `input.length` is `currStart + result3len` to see if provided `input` is **exactly** one `jpakePassivePayload` instance and nothing more.

Reply:
@lsolodkova

Reply:
Also relevant for other `fromBytes` methods.

Reply:
Done.